pilot/pkg/serviceregistry/serviceentry/store.go

```diff
@@ -66,20 +69,21 @@ func (s *serviceInstancesStore) deleteInstances(key configKey, instances []*mode
 func (s *serviceInstancesStore) addInstances(key configKey, instances []*model.ServiceInstance) {
 	for _, instance := range instances {
 		ikey := makeInstanceKey(instance)
+		hostPort := ikey.hostname.String() + "|" + strconv.Itoa(instance.ServicePort.Port)
```

Comment:
nit: a struct of {hostname,port} would be more efficient I think. The set is generic so it can handle it without problem

Reply:
good point. I will change